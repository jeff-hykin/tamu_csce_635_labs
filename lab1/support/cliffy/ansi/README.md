# Ansi

<p align="center" class="badges-container">
  <a href="https://github.com/c4spar/deno-cliffy/releases">
    <img alt="Version" src="https://img.shields.io/github/v/release/c4spar/deno-cliffy?logo=github&color=blue" />
  </a>
  <a href="https://github.com/c4spar/deno-cliffy/actions/workflows/test.yml">
    <img alt="Build status" src="https://github.com/c4spar/deno-cliffy/workflows/Test/badge.svg?branch=main" />
  </a>
  <a href="https://codecov.io/gh/c4spar/deno-cliffy">
    <img src="https://codecov.io/gh/c4spar/deno-cliffy/branch/main/graph/badge.svg"/>
  </a>
  <a href="https://github.com/c4spar/deno-cliffy/labels/module%3Aansi-escape">
    <img alt="issues" src="https://img.shields.io/github/issues/c4spar/deno-cliffy/module:ansi-escape?label=issues&logo=github&color=yellow">
  </a>
  <a href="https://deno.land/">
    <img alt="Deno version" src="https://img.shields.io/badge/deno-^1.4.0-blue?logo=deno" />
  </a>
  <a href="https://doc.deno.land/https/deno.land/x/cliffy/ansi/mod.ts">
    <img alt="doc" src="https://img.shields.io/badge/deno-doc-yellow?logo=deno" />
  </a>
  <a href="https://discord.gg/ghFYyP53jb">
    <img alt="Discord" src="https://img.shields.io/badge/join-chat-blue?logo=discord&logoColor=white" />
  </a>
  <a href="../LICENSE">
    <img alt="License" src="https://img.shields.io/github/license/c4spar/deno-cliffy?logo=github" />
  </a>
  <br>
  <a href="https://deno.land/x/cliffy">
    <img alt="Discord" src="https://img.shields.io/badge/Published on deno.land-blue?logo=deno&logoColor=959DA6&color=272727" />
  </a>
</p>

The documentation has been moved to https://cliffy.io/docs/ansi.
