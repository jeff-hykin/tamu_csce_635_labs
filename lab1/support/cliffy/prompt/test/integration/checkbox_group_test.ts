import { ansi } from "../../../ansi/ansi.ts";
import { snapshotTest } from "../../../testing/snapshot.ts";
import { Checkbox } from "../../checkbox.ts";

await snapshotTest({
  name: "checkbox prompt with groups",
  meta: import.meta,
  osSuffix: ["windows"],
  steps: {
    "should select a group option": {
      stdin: ansi
        .cursorDown
        .cursorDown
        .text(" ")
        .text("\n")
        .text("\n")
        .toArray(),
    },
    "should select a child option": {
      stdin: ansi
        .cursorDown
        .cursorDown
        .cursorForward
        .cursorDown
        .text(" ")
        .text("\n")
        .text("\n")
        .toArray(),
    },
  },
  async fn() {
    await Checkbox.prompt({
      message: "Select an option",
      options: [
        { name: "Foo", value: "foo" },
        { name: "Bar", value: "bar" },
        {
          name: "Baz",
          options: [
            { name: "Beep", value: "beep" },
            { name: "Boop", value: "boop" },
          ],
        },
      ],
    });
  },
});
