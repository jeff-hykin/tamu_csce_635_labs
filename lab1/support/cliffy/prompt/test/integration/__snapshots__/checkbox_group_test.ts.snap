export const snapshot = {};

snapshot[`checkbox prompt with groups > should select a group option 1`] = `
stdout:
"? Select an option
❯ ✘ Foo
  ✘ Bar
  ✘ 📁 Baz\\x1b[3A\\x1b[0G\\x1b[?25l\\x1b[G\\x1b[0J
? Select an option
  ✘ Foo
❯ ✘ Bar
  ✘ 📁 Baz\\x1b[3A\\x1b[0G\\x1b[?25l\\x1b[G\\x1b[0J
? Select an option
  ✘ Foo
  ✘ Bar
❯ ✘ 📁 Baz\\x1b[3A\\x1b[0G\\x1b[?25l\\x1b[G\\x1b[0J
? Select an option
  ✘ Foo
  ✘ Bar
❯ ✔ 📁 Baz\\x1b[3A\\x1b[0G\\x1b[?25l\\x1b[G\\x1b[0J
? Select an option
  ✘ Foo
  ✘ Bar
❯ ✔ 📁 Baz
Press ↵ again to submit. To open the selected group press →.\\x1b[4A\\x1b[0G\\x1b[?25l\\x1b[G\\x1b[0J
? Select an option › beep, boop
\\x1b[?25h\\x1b[?25h"
stderr:
""
`;

snapshot[`checkbox prompt with groups > should select a child option 1`] = `
stdout:
"? Select an option
❯ ✘ Foo
  ✘ Bar
  ✘ 📁 Baz\\x1b[3A\\x1b[0G\\x1b[?25l\\x1b[G\\x1b[0J
? Select an option
  ✘ Foo
❯ ✘ Bar
  ✘ 📁 Baz\\x1b[3A\\x1b[0G\\x1b[?25l\\x1b[G\\x1b[0J
? Select an option
  ✘ Foo
  ✘ Bar
❯ ✘ 📁 Baz\\x1b[3A\\x1b[0G\\x1b[?25l\\x1b[G\\x1b[0J
? Select an option
  ✘ 📂 Baz
 ❯ ✘ Beep
   ✘ Boop\\x1b[3A\\x1b[0G\\x1b[?25l\\x1b[G\\x1b[0J
? Select an option
  ✘ 📂 Baz
   ✘ Beep
 ❯ ✘ Boop\\x1b[3A\\x1b[0G\\x1b[?25l\\x1b[G\\x1b[0J
? Select an option
  ◉ 📂 Baz
   ✘ Beep
 ❯ ✔ Boop\\x1b[3A\\x1b[0G\\x1b[?25l\\x1b[G\\x1b[0J
? Select an option
  ◉ 📂 Baz
   ✘ Beep
 ❯ ✔ Boop
Press ↵ again to submit. To check or uncheck the selected option press space.\\x1b[4A\\x1b[0G\\x1b[?25l\\x1b[G\\x1b[0J
? Select an option › Boop
\\x1b[?25h\\x1b[?25h"
stderr:
""
`;
